src/main/java/frc/robot/subsystems/HandlerSubsystem.java

```diff
@@ -13,84 +16,166 @@
 public class HandlerSubsystem extends SubsystemBase {
   private enum State {
     EMPTY,
-    INTAKING,
-    LOADING,
-    LOADED,
-    EJECTING,
-    CANCELLING
+    INTAKING_CORAL,
+    INTAKING_ALGAE,
+    LOADING_CORAL,
+    LOADED_CORAL,
+    LOADED_ALGAE,
+    EJECTING_CORAL,
+    EJECTING_ALGAE,
+    CANCELLING_CORAL,
+    CANCELLING_ALGAE
   }
 
   private State m_state = State.EMPTY;
   private final SparkMax m_motor;
   private final RelativeEncoder m_encoder;
+  private final Pololu4079 m_algaeProxSensor;
+  private final Pololu4079 m_backProxSensor;
+  private final Pololu4079 m_frontProxSensor;
+  private final Pololu4079 m_distanceSensor;
   private final SendableChooser<State> m_chooser = new SendableChooser<>();
 
-  private final double m_intakeSpeed = HandlerConstants.kIntakeSpeed;
-  private final double m_ejectSpeed = HandlerConstants.kEjectSpeed;
+  private final double m_intakeSpeedCoral = HandlerConstants.kIntakeSpeedCoral;
+  private final double m_ejectSpeedCoral = HandlerConstants.kEjectSpeedCoral;
+  private final double m_intakeSpeedAlgae = HandlerConstants.kIntakeSpeedAlgae;
+  private final double m_ejectSpeedAlgae = HandlerConstants.kEjectSpeedAlgae;
```

Comment:
Let's get rid of these and directly use the constants. There's no good reason to make copies of them, certainly not as member variables.